apps/src/applab/setPropertyDropdown.js

```diff
@@ -48,137 +54,183 @@ var fullDropdownOptions = _.uniq(Object.keys(PROP_INFO).map(function (key) {
   return '"' + PROP_INFO[key].friendlyName + '"';
 }));
 
-// Which of the items from PROP_INFO should each element type use
-var PROP_NAMES = {};
-PROP_NAMES[ElementType.BUTTON] = [
-  'text',
-  'width',
-  'height',
-  'x',
-  'y',
-  'textColor',
-  'backgroundColor',
-  'fontSize',
-  'textAlign',
-  'image',
-  'hidden'
-];
-PROP_NAMES[ElementType.TEXT_INPUT] = [
-  'placeholder',
-  'width',
-  'height',
-  'x',
-  'y',
-  'textColor',
-  'backgroundColor',
-  'fontSize',
-  'textAlign',
-  'hidden'
-];
-PROP_NAMES[ElementType.LABEL] = [
-  'text',
-  'width',
-  'height',
-  'x',
-  'y',
-  'textColor',
-  'backgroundColor',
-  'fontSize',
-  'textAlign',
-  'hidden'
-];
-PROP_NAMES[ElementType.DROPDOWN] = [
-  'options',
-  'width',
-  'height',
-  'x',
-  'y',
-  'textColor',
-  'backgroundColor',
-  'fontSize',
-  'textAlign',
-  'hidden'
-];
-PROP_NAMES[ElementType.RADIO_BUTTON] = [
-  'groupId',
-  'width',
-  'height',
-  'x',
-  'y',
-  'hidden',
-  'checked'
-];
-PROP_NAMES[ElementType.CHECKBOX] = [
-  'width',
-  'height',
-  'x',
-  'y',
-  'hidden',
-  'checked'
-];
-PROP_NAMES[ElementType.IMAGE] = [
-  'width',
-  'height',
-  'x',
-  'y',
-  'picture',
-  'hidden'
-];
-PROP_NAMES[ElementType.CANVAS] = [
-  'canvasWidth',
-  'canvasHeight',
-  'x',
-  'y',
-];
-PROP_NAMES[ElementType.SCREEN] = [
-  'backgroundColor',
-  'screenImage'
-];
-PROP_NAMES[ElementType.TEXT_AREA] = [
-  'text',
-  'width',
-  'height',
-  'x',
-  'y',
-  'textColor',
-  'backgroundColor',
-  'fontSize',
-  'textAlign',
-  'readonly',
-  'hidden'
-];
-PROP_NAMES[ElementType.CHART] = [
-  'width',
-  'height',
-  'x',
-  'y',
-  'hidden'
-];
-PROP_NAMES[ElementType.SLIDER] = [
-  'width',
-  'height',
-  'x',
-  'y',
-  'value',
-  'min',
-  'max',
-  'step',
-  'hidden'
-];
+/**
+ * Information about properties pertaining to each element type. Values have the following
+ * fields. The latter two fields (friendlyNames and infoForFriendlyNames) are initialized
+ * through code run at startup time.
+ * propertyNames: a list of names (keys for PROP_INFO) associated with that element
+ * dropdownOptions: an array of property friendly names to be shown to the user for that element
+ * infoForFriendlyName: map from all friendly names for that element to property info
+ */
+var PROPERTIES = {};
+PROPERTIES[ElementType.BUTTON] = {
+  propertyNames: [
+    'text',
+    'width',
+    'height',
+    'x',
+    'y',
+    'textColor',
+    'backgroundColor',
+    'fontSize',
+    'textAlign',
+    'image',
+    'hidden'
+  ]
+};
+PROPERTIES[ElementType.TEXT_INPUT] = {
+  propertyNames: [
+    'text',
+    'placeholder',
+    'width',
+    'height',
+    'x',
+    'y',
+    'textColor',
+    'backgroundColor',
+    'fontSize',
+    'textAlign',
+    'hidden'
+  ]
+};
+PROPERTIES[ElementType.LABEL] = {
+  propertyNames: [
+    'text',
+    'text',
```

Comment:
I have no idea how that got there, but I did see it earlier and thought I removed it. Clearly, I need to improve by git-foo.